sycl-fusion/jit-compiler/lib/helper/ConfigHelper.h

```diff
@@ -11,11 +11,44 @@
 
 #include "Options.h"
 
+#include <memory>
+#include <unordered_map>
+
 namespace jit_compiler {
 
+class Config {
+
+public:
+  template <typename Opt> typename Opt::ValueType get() const {
```

Comment:
Not a change for this PR, but I don't think we should return a default initialized `T` here in case there is a miss. Assuming the default option is the default initialized option is not good. Users might rely on this and get a non-default option. Either:
1. Return some `Opt::default()` on miss.
1. Change interface, always returning an iterator here that can be `end()` in case this is not found

Reply:
Agreed, I'll file a ticket for this.